addons/mass_mailing/tests/test_mailing_list.py

```diff
@@ -155,3 +155,89 @@ def test_mailing_list_merge_cornercase(self):
         merge = merge_form.save()
         self.assertEqual(merge.src_list_ids, self.mailing_list_1 + self.mailing_list_2)
         self.assertEqual(merge.dest_list_id, self.mailing_list_3)
+
+
+class TestMailingContactImport(MassMailCommon):
+    """Test the transient <mailing.contact.import>."""
+
+    @users('user_marketing')
+    def test_mailing_contact_import(self):
+        first_list, second_list, third_list = self.env['mailing.list'].create([
+            {'name': 'First mailing list'},
+            {'name': 'Second mailing list'},
+            {'name': 'Third mailing list'},
+        ])
+
+        self.env['mailing.contact'].create([
+            {
+                'name': 'Already Exists',
+                'email': 'already_exists@example.com',
+                'list_ids': first_list.ids,
+            }, {
+                'email': 'already_exists_in_second_list@example.com',
+                'list_ids': second_list.ids,
+            }
+        ])
+
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'Test',
+            'subject': 'Test',
+            'contact_list_ids': (first_list | second_list).ids,
+        })
+
+        contact_import = Form(self.env['mailing.contact.import'].with_context(
+            active_id=mailing.id,
+            active_model='mailing.mailing',
+        ))
+
+        contact_import.contact_list = '''
+            alice@example.com
+            bob@example.com
+            "Bob" <bob@EXAMPLE.com>
+            "Test" <bob@example.com>
+            already_exists@example.com
+            already_exists_in_second_list@example.com
```

Comment:
Just for testing, couldn't we also check for "already_exist_in_both" just to check with a contact already in all lists ?